sklearn/model_selection/_split.py

```diff
@@ -891,6 +901,8 @@ def get_n_splits(self, X, y, groups):
         """
         if groups is None:
             raise ValueError("The groups parameter should not be None")
+        groups = check_array(groups, ensure_2d=False, dtype=None)
+        X, y, groups = indexable(X, y, groups)
```

Comment:
Hmm... Isn't that expensive? We could use `sum(1 for _ in self.split(...)` to save space but still it would require computation of splits...

Besides, if we have a specific way of getting the `n_splits` for each cross-validator without enumerating the splits, what would be the advantage of such a default method?